EIPS/eip-2315.md

```diff
@@ -0,0 +1,91 @@
+---
+eip: 2315
+title: Simple Subroutines for the EVM
+status: Draft
+type: Standards Track
+category: Core
+author: Greg Colvin
+discussions-to: https://ethereum-magicians.org/t/eip-2315-simple-subroutines-for-the-evm/3941
+created: 2019-10-17
+---
+## Abstract
+
+This proposal introduces two opcodes to support subroutines:  `JUMPSUB` and `RETSUB`.
+
+## Motivation
+
+The EVM does not provide subroutines as a primitive.  Instead, calls must be synthesized by fetching and pushing the current program counter on the data stack and jumping to the subroutine address; returns must be synthesized by getting the return address to the top of stack and jumping back to it.
+
+## Specification
+
+**_JUMPSUB_**
+
+* Jumps to the address on top of the stack.  This must be a `JUMPDEST`.
```

Comment:
The word `address` is ambiguous, at to a casual reader. I'd suggest `pc` or `offset` to remove any ambiguity.

Reply:
Gotcha.

Reply:
Changed to:
> Jumps to the address on top of the stack, which must be the offset of a `JUMPDEST`.
'Address and program-counter are used with that meaning in the previous paragraph, so this is a good place for a reminder.